pkg/apis/core/validation/validation.go

```diff
@@ -4362,17 +4362,25 @@ func validatePodResources(spec *core.PodSpec, podClaimNames sets.Set[string], fl
 	if spec.Resources == nil {
 		return nil
 	}
+	resourcesFldPath := fldPath.Child("resources")
```

Comment:
(no action item here, just documenting my thought process)

Normally when we change validation, we need to consider [ratcheting validation](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#ratcheting-validation) to support n-1 version skew betwen api servers.

But in this case since it's alpha in 1.33 and the Pod.spec.resources field is immutable in 1.33, then this is a safe change to make. Objects created in 1.34 will have the tightened validation, and 1.33 api servers will still succeed validation with the looser validation.